Schedule/src/main/java/com/sps/app/servlets/EventServlet.java

```diff
@@ -0,0 +1,188 @@
+
+package com.sps.app.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.List;
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.KeyRange;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Query.Filter; 
+import com.google.appengine.api.datastore.Query.FilterPredicate; 
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Map;
+import java.util.Date;
+import java.util.Locale;
+import java.text.*;
+import com.sps.app.Event;
+import java.lang.*;
+
+@WebServlet("/")
+public class EventServlet extends HttpServlet {
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    String userId = getParameter(request, "userId","");
+    if(userId == null){
+      response.setStatus(400);
+      response.getWriter().println("userId cannot be empty");
+      return;
+    }
+
+    List<Event> eventList = new ArrayList<>();
+    Query query = new Query("Event");
+    PreparedQuery results = datastore.prepare(query);
+    SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.ENGLISH);
+
+    for (Entity entity : results.asIterable()) {
+      String entityUser = (String) entity.getProperty("userId");
+      if(entityUser.equals(userId)){
+        String title = (String) entity.getProperty("title");
+        String startTime = (String)entity.getProperty("startTime");
+        String endTime = (String)entity.getProperty("endTime");
+        String content = (String) entity.getProperty("content");
+
+        Event event = new Event(userId, title, startTime, endTime, content);
+        eventList.add(event);
+      }
+    }
+
+    String json = convertToJsonUsingGson(eventList);
+    response.setContentType("application/json; charset=UTF-8");
+    response.getWriter().println(json);
+  }
+    
+  private String convertToJsonUsingGson(List<Event> eventList) {
+    Gson gson = new Gson();
+    String json = gson.toJson(eventList);
+    return json;
+  }
+    
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    String userId = getParameter(request, "userId", "");
```

Comment:
Just wondering, why are you using getParameter method instead of request.getParameter? I thought the walkthroughs write it this way to show us how it works?